safe-global/safe-wallet-web
# feat: untrusted transfer transaction warnings

## What it solves

Resolves #2991 

## How this PR fixes it
- Fetches all transactions from the CGW
- Displays untrusted txs without a nonce opaque
- Shows a warning icon for untrusted tokens in transfer transactions
- Shows a new modal when copying addresses from untrusted addresses to prevent address poisoning attacks.

## Todo
- [x] Track copying / copy attempts

## New Analytics events
- Show copy address warning
- Proceed and copy address

## How to test it
- Open a Safe that received Spam / scam tokens
- Check that the warnings are displayed and the untrusted txs are shown opaque
- Try to copy any address involved with an untrusted transaction and observe the new modal

## Screenshots
![Screenshot 2023-12-11 at 18 23 49](https://github.com/safe-global/safe-wallet-web/assets/2670790/9180efe2-23df-4680-bf4f-7697e2ce29f8)
![Screenshot 2023-12-11 at 16 46 11](https://github.com/safe-global/safe-wallet-web/assets/2670790/a476e7a1-d4ee-46d1-8603-b68540cb056a)

## Checklist
